sr_speech_control/src/sr_speech_control/speech_control.py

```diff
@@ -15,66 +15,46 @@
 # with this program. If not, see <http://www.gnu.org/licenses/>.
 
 import rospy
-import time
 import speech_recognition as sr
-from difflib import get_close_matches
 from std_msgs.msg import String
 
 
 class SpeechControl(object):
-    def __init__(self, trigger_word, command_words, command_topic='speech_control',
-                 non_speaking_duration=0.1, pause_threshold=0.2):
+    def __init__(self, trigger_word, command_topic='sr_speech_control'):
+        self.microphone = sr.Microphone()
+        for i, microphone_name in enumerate(sr.Microphone.list_microphone_names()):
+            if "pulse" in microphone_name:
```

Comment:
Any way to set the pulseaudio as a default microphjone in the docker's system? That way we maintain the flexibility of having this software using a default mike, instead of hardcoding a specific workaround that would be used only in docker. If that if not possible, I'd rather parmetrize this to have the node take a parameters like "use_pulse" or "docker" so we dont hard-code a workaround for docker only.

Reply:
I have added a command line parameter for "preferred microphone" in 47481e57550e3a76ae6a454e2e000c7f9da3dbdf. Do you think it is good enough?

Reply:
Since its a rosnode, do you mind using private ros params for this (rospy.get_param)? That way we can use it cleanly using ros nomenclarute in launchfiles (<param name="some_param" value="$(arg package_name)" />).

Reply:
Also, would be good to leave an option to just use default microphone if user desires so, instead of always having to pick one. So you either just do `self.microphone = sr.Microphone()` to just choose default or you pass the param and find spefic desired microphone using the loop. What do you think?

Reply:
Changed to use private ROS parameters in 6f4378e64a161620611a9c43141f84b96cb6fdd9. Default microphone is now used by default.